client/js/libs/handlebars/parse.js

```diff
@@ -2,125 +2,111 @@
 
 const Handlebars = require("handlebars/runtime");
 const URI = require("urijs");
+const parseStyle = require("./ircmessageparser/parseStyle");
+const findChannels = require("./ircmessageparser/findChannels");
+const merge = require("./ircmessageparser/merge");
 
-module.exports = function(text) {
-	text = Handlebars.Utils.escapeExpression(text);
-	text = colors(text);
-	text = channels(text);
-	text = uri(text);
-	return text;
-};
+const commonSchemes = [
+	"http", "https",
+	"ftp", "sftp",
+	"smb", "file",
+	"irc", "ircs",
+	"svn", "git",
+	"steam", "mumble", "ts3server",
+	"svn+ssh", "ssh",
+];
+
+function findLinks(text) {
+	let result = [];
+	let lastPosition = 0;
+
+	URI.withinString(text, function(url, start, end) {
+		// v-- fix: url was modified and does not match input string -> cant be mapped
+		if (text.indexOf(url, lastPosition) < 0) {
+			return;
+		}
+		// ^-- /fix: url was modified and does not match input string -> cant be mapped
```

Comment:
Sorry I meant is it still broken in URI.js? Meaning, is this still needed pending a fix in URI.js or is it now unnecessary to keep in our own codebase?